packages/react/src/utils/mergeThemes.ts

```diff
@@ -42,6 +43,26 @@ export const emptyTheme: ThemePrepared = {
 // Component level merge functions
 // ----------------------------------------
 
+const mergeComponentStylesCache = {}
+
+export const mergeComponentStylesWithCache = (
+  hashObj,
+  sources: (ComponentSlotStylesInput | null | undefined)[],
+) => {
+  try {
+    // using the hash fn is costly... is stringify valid? [I previosly convert all possible fn to string...]
+    // const hashVal = hash(hashObj)
+    const hashVal = JSON.stringify(hashObj)
```

Comment:
Using the hash fn was constly, the running time was doubled compard to master :S With using only strigify the perfs are actually the same (no gains as per my tests) @levithomason any further suggestions/proposals?